plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/parser/DockerImageIdentifierParser.java

```diff
@@ -26,59 +26,74 @@
  */
 public class DockerImageIdentifierParser {
 
-    private static final String HOSTNAME_COMPONENT = "(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])";
-    private static final String REGISTRY           = HOSTNAME_COMPONENT + "(?:\\." + HOSTNAME_COMPONENT + ")*([:][0-9]+)?";
-    private static final String CAPTURED_REGISTRY  = "(?<registry>" + REGISTRY + ")";
-    private static final String SEPARATOR           = "(?:[_.]|__|[-]*)";
-    private static final String ALPHA_NUMERIC       = "[a-z0-9]+";
-    private static final String NAME_COMPONENT      = ALPHA_NUMERIC + "(?:" + SEPARATOR + ALPHA_NUMERIC + ")*";
-    private static final String REPOSITORY          = NAME_COMPONENT + "(?:[/]" + NAME_COMPONENT + ")*";
-    private static final String CAPTURED_REPOSITORY = "(?<repository>" + REPOSITORY + ")";
-    private static final String NAME                = "(?:" + CAPTURED_REGISTRY + "[/])?" + CAPTURED_REPOSITORY;
-
-    private static final Pattern IMAGE_PATTERN = Pattern.compile(NAME);
-
+    // Validation rules are taken from https://github.com/docker/distribution/blob/master/reference/regexp.go
+    private static final String  HOSTNAME_COMPONENT = "(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])";
+    private static final String  REGISTRY           = HOSTNAME_COMPONENT + "(?:\\." + HOSTNAME_COMPONENT + ")*(?::[0-9]+)?";
+    private static final String  SEPARATOR          = "(?:[._]|__|[-]*)";
+    private static final String  ALPHA_NUMERIC      = "[a-z0-9]+";
+    private static final String  NAME_COMPONENT     = ALPHA_NUMERIC + "(?:" + SEPARATOR + ALPHA_NUMERIC + ")*";
+    private static final String  REPOSITORY         = NAME_COMPONENT + "(?:/" + NAME_COMPONENT + ")*";
+    private static final String  TAG                = "[\\w][\\w.-]*";
+    private static final String  DIGEST             = "[\\w+.:-]+";
+    private static final String  NAME               = "(?:" + REGISTRY + "/)?" + REPOSITORY;
+    private static final String  REFERENCE          = NAME + "(?::" + TAG + ")?" + "(?:@" + DIGEST + ")?";
+    private static final Pattern IMAGE_PATTERN      = Pattern.compile(REFERENCE);
 
+    /**
+     * Validates and parse docker image reference into object that holds reference components
+     *
+     * @param image image reference to parse
+     * @throws DockerFileException
+     * @throws IllegalArgumentException
```

Comment:
There is no `IllegalArgumentException` thrown + `DockerFileExpcetion` is not documented, please do it.